tests/test_zero/test_stateful_tensor_mgr.py

```diff
@@ -27,12 +27,16 @@ def __init__(self) -> None:
         self.p2 = Parameter(torch.empty(1024, 1024, 128))
 
 
-def run_stm():
-    cuda_capacity = colo_device_memory_capacity(get_current_device())
-    fraction = (1.4 * 1024**3) / cuda_capacity
-    # limit max memory to 1.4GB
-    # which means only 2 parameters can be on CUDA
+def limit_cuda_memory(memory_in_g: float):
+    cuda_capacity = torch.cuda.get_device_properties(get_current_device()).total_memory
+    fraction = (memory_in_g * 1024**3) / cuda_capacity
     colo_set_process_memory_fraction(fraction)
+
+
+def run_stm():
+    # warmup phase use 20% CUDA memory to store params
+    # only 2 params can be on CUDA
+    limit_cuda_memory(5.1)
```

Comment:
why 5.1?

Reply:
Because we set CUDA memory limit by a fraction which is a floating point, and this leads to precision error. So we set it to a larger value.